From 90c51fb391b7214438292cbc8d8589f2ddd08a7d Mon Sep 17 00:00:00 2001
From: Zeng Heng <zengheng4@huawei.com>
Date: Tue, 24 Sep 2024 21:47:52 +0800
Subject: [PATCH 682/960] downstream: Revert "arm64: head.S: Initialise MPAM
 EL2 registers and disable traps"

This reverts commit a01fda352718f178db2effe5e9d2c1b6a5602356.

If BIOS firmware doesn't enable MPAM function under EL3 environment, while
the hardware has the MPAM ability, that would cause illegal instruction
fault when access MPAM registers.

It should be noted that this will bring potential risks to Kdump. If the
previous kernel has an unhandled MPAM exception interrupt, it would cause
Kdump crash after re-enable interrupt capability.

Fixes: a01fda352718 ("arm64: head.S: Initialise MPAM EL2 registers and disable traps")
Signed-off-by: Zeng Heng <zengheng4@huawei.com>
Signed-off-by: Liu Shixin <liushixin2@huawei.com>
---
 arch/arm64/include/asm/el2_setup.h | 16 ----------------
 1 file changed, 16 deletions(-)

diff --git a/arch/arm64/include/asm/el2_setup.h b/arch/arm64/include/asm/el2_setup.h
index b50b9825d60b..6f65322e2fd0 100644
--- a/arch/arm64/include/asm/el2_setup.h
+++ b/arch/arm64/include/asm/el2_setup.h
@@ -302,21 +302,6 @@
 	msr	spsr_el2, x0
 .endm
 
-.macro __init_el2_mpam
-#ifdef CONFIG_ARM64_MPAM
-	/* Memory Partioning And Monitoring: disable EL2 traps */
-	mrs	x1, id_aa64pfr0_el1
-	ubfx	x0, x1, #ID_AA64PFR0_EL1_MPAM_SHIFT, #4
-	cbz	x0, .Lskip_mpam_\@		// skip if no MPAM
-	msr_s	SYS_MPAM2_EL2, xzr		// use the default partition
-						// and disable lower traps
-	mrs_s	x0, SYS_MPAMIDR_EL1
-	tbz	x0, #17, .Lskip_mpam_\@		// skip if no MPAMHCR reg
-	msr_s	SYS_MPAMHCR_EL2, xzr		// clear TRAP_MPAMIDR_EL1 -> EL2
-.Lskip_mpam_\@:
-#endif /* CONFIG_ARM64_MPAM */
-.endm
-
 /**
  * Initialize EL2 registers to sane values. This should be called early on all
  * cores that were booted in EL2. Note that everything gets initialised as
@@ -334,7 +319,6 @@
 	__init_el2_stage2
 	__init_el2_gicv3
 	__init_el2_hstr
-	__init_el2_mpam
 	__init_el2_nvhe_idregs
 	__init_el2_cptr
 	__init_el2_fgt
-- 
2.43.0

